### ISSUE TITLE
Keyboard focus is lost after invoking 'Esc' key in 'Search' dialog.

### ISSUE THREAD

**Issue by msftedad** [facebook/docusaurus#9164]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [ ] I have tried the `npm run clear` or `yarn clear` command.
- [ ] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [ ] I have read the console error message carefully (if applicable).

### Description

Keyboard focus is lost on invoking esc key in search dialog.

On invoking Esc key, keyboard focus is getting lost.

After Activating Esc key, keyboard focus should land on the last interacted element i.e., on ‘Search’ button.

User with motor impairment who rely on keyboard will face if keyboard focus resets.

[incorrect keyboard focus .webm](https://github.com/facebook/docusaurus/assets/93735775/466e790d-72f7-4951-9dfd-2277cad86733)

### Steps to reproduce

**Repro Steps**:
1. Open https://www.fast.design/ URL.
2. Navigate to 'Search' button and invoke it.
3. Now press Esc key to close the ‘Search’ dialog and observe the keyboard focus order.

### Expected behavior

After hitting Esc key, keyboard focus should land on the last interacted element i.e., on ‘Search’ button.

### Actual behavior

Keyboard focus is lost on invoking esc key in search dialog.

### Your environment

- Public source code:
- Public site URL: https://www.fast.design/
- Docusaurus version used: Version: 2.4.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Microsoft Edge
**Version** 114.0.1823.86 (Official build) (64-bit)
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): **OS Name**: Microsoft Windows 11 Enterprise
**Version**: 10.0.22621 Build 22621

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Hi, the search dialog is not maintained by us but by Algolia, so this issue should be reported to them.

cc @shortcuts

**Comment by msftedad** [comment]
@Josh-Cena, Hi we have connected with po team and we got a below reply

We have verified the issue in Algolia and observed we are not getting the similar kind of UI. For search Algolia no hint is defined. Please let us know correct product or UI to repro and file a new bug.

Can you please check and let us know. Thanks.

**Comment by slorber** [comment]
I'm not sure if this focus management should be in Algolia or the Docusaurus integration. Until we figure out we can keep this open.

**Comment by slorber** [comment]
@shortcuts I assume that since you also maintain the search button that opens the mobile, it should more likely be your responsibility to restore focus to that button?

**Comment by shortcuts** [comment]
hey @slorber, it definitely makes sense to have this one the Algolia side. Those kind of UI issues should be reported via https://github.com/algolia/docsearch/issues/new/choose so that the team knows about it, not sure if you can transfer it?

cc @shaneafsar for visibility

**Comment by slorber** [comment]
Thanks

I don't think I can transfer it no.

@msftedad can you re-open the issues on this Algolia DocSearch repo?

Docusaurus scope does not include:
- The Search button
- The Search modal

**Comment by msftedad** [comment]
@slorber.Thank you for the information. We will reopen this issue in Algolia.
